main.c

```diff
@@ -540,15 +550,29 @@ void USCI_B1_ISR (void)
                 break;
             case LED3_OFF:
 		hal_ext_led(3, 0);
-                break;            
+                break;
+            case UARTMUXSEL_SET_MIKROBUS:
+                uartmuxState = UARTMUXSEL_MODE_MIKROBUS;
+                hal_ext_uart(false);
+                hal_ext_led(3, 0);
+                break;
+            case UARTMUXSEL_SET_MSP430:
+                uartmuxState = UARTMUXSEL_MODE_MSP430;
+                hal_ext_uart(true);
+                hal_ext_led(3, 1);
+                break;                 
             default:
                 break;
             }
             break;
         }
         case USCI_I2C_UCTXIFG:
```

Comment:
Is this state within `switch (USB_getConnectionState())`? Why would USB_getConnectionState() provide any state relative to I2C? Shouldn't this just be another test within the while(1) loop?